client/my-sites/email/email-forwarding/email-forwarding-add-new-compact.jsx

```diff
@@ -0,0 +1,219 @@
+import { Button, Gridicon } from '@automattic/components';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import FormButton from 'calypso/components/forms/form-button';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormInputValidation from 'calypso/components/forms/form-input-validation';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import FormTextInputWithAffixes from 'calypso/components/forms/form-text-input-with-affixes';
+import { validateAllFields } from 'calypso/lib/domains/email-forwarding';
+import formState from 'calypso/lib/form-state';
+
+class EmailForwardingAddNewCompact extends Component {
+	static propTypes = {
+		fields: PropTypes.object,
+		index: PropTypes.number,
+		removeHandler: PropTypes.func.isRequired,
+		selectedDomainName: PropTypes.string.isRequired,
+		updateHandler: PropTypes.func.isRequired,
+		emailForwards: PropTypes.array,
+	};
+
+	state = {
+		formSubmitting: false,
+		fields: this.props.fields,
+	};
+
+	getInitialFields() {
+		return this.props.fields;
+	}
+
+	getInitialFormState() {
+		return {
+			formSubmitting: false,
+			fields: this.props.fields,
+		};
+	}
+
+	UNSAFE_componentWillMount() {
+		this.formStateController = formState.Controller( {
+			initialFields: this.getInitialFields(),
+			onNewState: this.setFormState,
+			validatorFunction: ( fieldValues, onComplete ) => {
+				onComplete( null, validateAllFields( fieldValues, this.props.emailForwards ?? [] ) );
+			},
+		} );
+
+		this.setFormState( this.formStateController.getInitialState() );
+	}
+
+	addNewEmailForwardClick = ( event ) => {
+		event.preventDefault();
+
+		if ( this.state.formSubmitting ) {
+			return;
+		}
+
+		this.setState( { formSubmitting: true } );
+
+		this.formStateController.handleSubmit( ( hasErrors ) => {
+			if ( hasErrors ) {
+				this.setState( { formSubmitting: false } );
+				return;
+			}
+
+			const { mailbox, destination } = formState.getAllFieldValues( this.state.fields );
+
+			this.props.addNewEmailForwardWithAnalytics(
+				this.props.selectedDomainName,
+				mailbox,
+				destination
+			);
+			this.formStateController.resetFields( this.getInitialFields() );
+			this.setState( { formSubmitting: false, showForm: true } );
+		} );
+	};
+
+	setFormState = ( fields ) => {
+		this.setState( { fields } );
+	};
+
+	addButton() {
```

Comment:
Nit: for this function, `removeButton`, and `formFields`, I think it would be better to prefix them with `render` to make it clear that we're rendering the buttons, not performing the add/remove action for a button.
```suggestion
	renderAddButton() {
```

Reply:
This `addButton()` is never called anywhere.

Reply:
I've renamed this function, but this is still a placeholder to support adding more than one email forward at a time. We're not trying to get that fully implemented in this PR, as the back end API only supports adding one forward at a time.

But I will update the PR description regardless.